### PR TITLE
prepare for RNC

### PR DESCRIPTION
**Pull request by pvinis**
# Summary
Renaming everything from `react-native-hooks` to `@react-native-community/hooks`.
This is in preparation for a new release under the RNC org.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [x] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by pvinis** [comment]
I'll wait on this until @dabit3 gives me the OK. I want him to be fully aware of this.

**Comment by dabit3** [comment]
Looks great @pvinis I appreciate you taking the initiative here and doing the work to make this happen.

**Comment by pvinis** [comment]
Since you are here, should we skip the release under the current name, and go for a release under RNC with all the latest changes?

**Comment by dabit3** [comment]
@pvinis I think that makes sense, but I will say do what you think is best based on your judgement.

**Comment by pvinis** [comment]
I'll merge this, and prepare for a `2.0.0`(?) release under RNC.

After that, we can use https://docs.npmjs.com/cli/deprecate to show a message to people that they should install `@react-native-community/hooks` instead. This can be done for the current package current version, from your account.

(thanks @lucasbento for the deprecation tip!)

File: README.md [hunk 1]
```diff
@@ -149,4 +149,4 @@ console.log('layout: ', layout)
 ```
 
 [version-badge]: https://img.shields.io/npm/v/react-native-hooks.svg?style=flat-square
-[package]: https://www.npmjs.com/package/react-native-hooks
\ No newline at end of file
+[package]: https://www.npmjs.com/package/react-native-hooks
```
Comment (lucasbento) [inline README.md]: Need to update this link as well 🤗

**Comment by pvinis** [comment]
Thanks both. This is ready. I'll keep it open until we finalize with @Titozzz on the release process, then I'll merge and do a release. Then I'll notify @dabit3 again for a deprecation.

Woop RNC 🙌

**Pull request merged**